src/ape/contracts/base.py

```diff
@@ -263,16 +263,16 @@ def __repr__(self) -> str:
         return self.abi.signature
 
     def serialize_transaction(self, *args, **kwargs) -> TransactionAPI:
-        if "sender" in kwargs and isinstance(kwargs["sender"], ContractInstance):
-            # Automatically impersonate contracts (if API available) when sender
-            kwargs["sender"] = self.account_manager.test_accounts[kwargs["sender"].address]
-
         kwargs = _convert_kwargs(kwargs, self.conversion_manager.convert)
         return self.provider.network.ecosystem.encode_transaction(
             self.address, self.abi, *args, **kwargs
         )
 
     def __call__(self, *args, **kwargs) -> ReceiptAPI:
+        if "sender" in kwargs and isinstance(kwargs["sender"], (ContractInstance, Address)):
+            # Automatically impersonate contracts (if API available) when sender
+            kwargs["sender"] = self.account_manager.test_accounts[kwargs["sender"].address]
+
```

Comment:
What's the reasoning for moving this? `serialize_transaction` gets called below

Reply:
to set kwargs["sender"] here to be ImpersonatedAccount so later if condition will work and use ImpersonatedAccount.call function

Reply:
So the flow here is that this same convert step was happening inside of `serialize_transaction`:

https://github.com/ApeWorX/ape/blob/2a82957dcbf97471fe5d2c6dc0b8359676aaaee9/src/ape/contracts/base.py#L276

Which is before the use of it to do `kwargs["sender"].call`:

https://github.com/ApeWorX/ape/blob/2a82957dcbf97471fe5d2c6dc0b8359676aaaee9/src/ape/contracts/base.py#L278-L279

Meaning they are functionally equivalent, except we loose the ability to call `serialize_transaction` directly and take advantage of this same feature